hackathon_site/hardware/serializers.py

```diff
@@ -46,9 +50,91 @@ def get_unique_hardware_count(obj: Category):
         return Hardware.objects.filter(categories__id=obj.id).count()
 
 
-class OrderSerializer(serializers.ModelSerializer):
+class OrderListSerializer(serializers.ModelSerializer):
     hardware_set = HardwareSerializer(many=True, read_only=True)
 
     class Meta:
         model = Order
         fields = ("id", "hardware_set", "team", "status", "created_at", "updated_at")
+
+
+class OrderCreateSerializer(serializers.Serializer):
+    class OrderCreateHardwareSerializer(serializers.Serializer):
+        id = serializers.PrimaryKeyRelatedField(
+            queryset=Hardware.objects.all(), many=False, required=True
+        )
+        quantity = serializers.IntegerField(required=True)
+
+    team_id = serializers.PrimaryKeyRelatedField(
```

Comment:
This doesn't need to be on the serializer imo. Since they have to be logged in to access the API, you can use `self.request.user.profile.team.id` in the view.

Cc @albogdan - do we think the tech team will ever be using the API to create orders for other teams? If not, this field can go. If the field stays, then we'll want a check that a regular user can't create an order except for the team they are on.

Reply:
Hmm now why did I put it in the description in the jira ticket.... What was past Graham thinking?

Reply:
Yea I thought it would be in the user profile too loool